pact/namespaces/ns.pact

```diff
@@ -35,21 +31,42 @@
       ( ns-name:string
         ns-admin:guard
         )
-    " Manages namespace install for Chainweb. Requires active row in registry \
+    " Manages namespace install for Chainweb. \
+    \ Supports principal namespaces for k: and w: \
+    \ using # instead of :. \
+    \ Non-principal namespaces require active row in registry \
     \ for NS-NAME with guard matching NS-ADMIN."
 
-    (validate-name ns-name)
+    (let*
+      ((parse-name
+         (lambda (n)
+           (let
+             ((take-two (take 2 n))
+              (drop-two (drop 2 n)))
+
+             (cond
+              ((= "k#" take-two)
+                (+ "k:" drop-two))
+              ((= "w#" take-two)
+                (+ "w:" (+ (take 43 drop-two) (+ ":" (drop 46 n)))))
```

Comment:
Since this is validation, should there be an enforce that the address is indeed long enough for this take and drop to be meaningful here?